src/pages/warnings/WarningSearchFilter.tsx

```diff
@@ -0,0 +1,93 @@
+import type { FC } from "react";
+import { memo } from "react";
+import { SearchAndFilter } from "@canonical/react-components";
+import type {
+  SearchAndFilterChip,
+  SearchAndFilterData,
+} from "@canonical/react-components/dist/components/SearchAndFilter/types";
+import { useSearchParams } from "react-router-dom";
+import {
+  paramsFromSearchData,
+  searchParamsToChips,
+} from "util/searchAndFilter";
+import { warningSeverities, warningStatuses } from "util/warningFilter";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchWarnings } from "api/warnings";
+
+export const QUERY = "query";
+export const STATUS = "status";
+export const SEVERITY = "severity";
+
+const QUERY_PARAMS = [QUERY, STATUS, SEVERITY];
+
+const WarningSearchFilter: FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const { data: warnings = [] } = useQuery({
+    queryKey: [queryKeys.warnings],
+    queryFn: async () => fetchWarnings(),
+    retry: false, // the api returns a 403 for users with limited permissions, surface the error right away
+  });
+
+  const projectSet = [
+    ...new Set(warnings.flatMap((warning) => warning.project)),
+  ];
```

Comment:
I'd suggest to completely drop the project filter, that simplifies this component and we avoid the issue with no projects being available.